itunes.sql

```diff
@@ -0,0 +1,647 @@
+CREATE TABLE IF NOT EXISTS `itunes`.`user's_playlist_info` (
+  `id` INT NOT NULL,
+  `user_id` INT NOT NULL,
+  `name` VARCHAR(45) NOT NULL,
+  PRIMARY KEY (`id`),
+  INDEX `fk_user's_playlist_user1_idx` (`user_id` ASC) VISIBLE,
+  CONSTRAINT `fk_user's_playlist_user1`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `itunes`.`user` (`id`)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`songs_saved_by_user`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`songs_saved_by_user` (
+  `song_id` INT NOT NULL,
+  `user_id` INT NOT NULL,
+  PRIMARY KEY (`song_id`, `user_id`),
+  INDEX `fk_song_has_user_user1_idx` (`user_id` ASC) VISIBLE,
+  INDEX `fk_song_has_user_song1_idx` (`song_id` ASC) VISIBLE,
+  CONSTRAINT `fk_song_has_user_song1`
+    FOREIGN KEY (`song_id`)
+    REFERENCES `itunes`.`song` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_song_has_user_user1`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `itunes`.`user` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`albums_saved_by_user`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`albums_saved_by_user` (
+  `album_id` INT NOT NULL,
+  `user_id` INT NOT NULL,
+  PRIMARY KEY (`album_id`, `user_id`),
+  INDEX `fk_album_has_user_user1_idx` (`user_id` ASC) VISIBLE,
+  INDEX `fk_album_has_user_album1_idx` (`album_id` ASC) VISIBLE,
+  CONSTRAINT `fk_album_has_user_album1`
+    FOREIGN KEY (`album_id`)
+    REFERENCES `itunes`.`album` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_album_has_user_user1`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `itunes`.`user` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`user_prefer_genre`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`user_prefer_genre` (
+  `user_id` INT NOT NULL,
+  `genre_id` INT NOT NULL,
+  PRIMARY KEY (`user_id`, `genre_id`),
+  INDEX `fk_user_has_genre_genre1_idx` (`genre_id` ASC) VISIBLE,
+  INDEX `fk_user_has_genre_user1_idx` (`user_id` ASC) VISIBLE,
+  CONSTRAINT `fk_user_has_genre_user1`
+    FOREIGN KEY (`user_id`)
+    REFERENCES `itunes`.`user` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE,
+  CONSTRAINT `fk_user_has_genre_genre1`
+    FOREIGN KEY (`genre_id`)
+    REFERENCES `itunes`.`genre` (`id`)
+    ON DELETE RESTRICT
+    ON UPDATE CASCADE)
+ENGINE = InnoDB;
+
+
+-- -----------------------------------------------------
+-- Table `itunes`.`_playlist_has_song`
+-- -----------------------------------------------------
+CREATE TABLE IF NOT EXISTS `itunes`.`_playlist_has_song` (
```

Comment:
is this even legal to start a table name with underscore in MySQL?

Reply:
Ok, I just checked and it is legal, but not a common practice.
However, you need to stick to your naming convention. In your database, no other table starts with an underscore, so please remove it here too.

Reply:
Changed